src/java.base/share/native/libjava/io_util.c

```diff
@@ -53,6 +53,48 @@ readSingle(JNIEnv *env, jobject this, jfieldID fid) {
     return ret & 0xFF;
 }
 
+jlong
+readN(JNIEnv *env, jobject this, jfieldID fid, jint len) {
+    if (len == 1) {
+        return readSingle(env, this, fid);
+    }
+    char stackBuf[len];
+    char *buf = stackBuf;
+
+    FD fd = getFD(env, this, fid);
+    if (fd == -1) {
+        JNU_ThrowIOException(env, "Stream Closed");
+        return -1;
+    }
+    jint nread = IO_Read(fd, buf, len);
+    if (nread == -1) {
+        JNU_ThrowIOExceptionWithLastError(env, "Read error");
```

Comment:
These methods are specified to throw the more specific I/O exception EOFException at EOF.

Reply:
You mean this must be changed to 
```c
JNU_ThrowByName(env, "java/io/EOFException", NULL);
```
?